### PR TITLE
EVG-2869 Implement project admin page subscriptions UI changes

### PR DESCRIPTION
**Pull request by richardsamuels**
Some images to show the workflow. This UI is right below the existing alerts UI:
project-subs1

Clicking the plus button spawns this:
project-subs3

Selecting a notification method looks something like this:
project-subs4

Saving it looks like this:
project-subs5

### CODE REVIEW

**Comment by richardsamuels** [comment]
Broader concern: i had to update angular-material because of a bug (https://github.com/angular/material/pull/10549). I'm a little scared of UI regressions, @john-m-liu any advice there?

File: public/static/js/projects.js [hunk 1]
```diff
@@ -444,6 +484,37 @@ mciModule.controller('ProjectCtrl', function($scope, $window, $http, $location)
...
+
+      $mdDialog.show(promise).then(function(data){
+          data.changed = true;
+          $scope.isDirty = true;
+          $scope.subscriptions[index] = data;
+      });
+  };
+
+  $scope.removeSubscription = function(index) {
+      if ($scope.subscriptions[index].id !== undefined) {
```
Comment (john-m-liu) [inline public/static/js/projects.js]: I would do ($scope.subscriptions[index] && $scope.subscriptions[index].id)

File: public/static/js/projects.js [hunk 2]
```diff
@@ -444,6 +484,37 @@ mciModule.controller('ProjectCtrl', function($scope, $window, $http, $location)
...
+          $scope.settingsFormData.delete_subscriptions.push($scope.subscriptions[index].id);
+      }
+      $scope.subscriptions = _.filter($scope.subscriptions, function(s, i) {
+          return index !== i;
+      });
+      $scope.isDirty = true;
+  };
+  $scope.show_build_break = true;
```
Comment (john-m-liu) [inline public/static/js/projects.js]: Don't think this is needed - it's always true

File: public/static/js/subscriptions.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+    $scope.closeDialog = function(save) {
+        if(save === true) {
+            subscriber = {
+                type: $scope.method.value,
+                target: $scope.targets[$scope.method.value],
+            }
+            subscriber.label = subscriberLabel(subscriber);
+
+            d = Object.assign({}, $scope.c.subscription);
```
Comment (john-m-liu) [inline public/static/js/subscriptions.js]: Don't think you need to copy the object here

File: public/static/js/subscriptions.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+
+        for (var i = 0; i < 64; i++)
+            text += possible.charAt(Math.floor(Math.random() * possible.length));
+
+        return text;
+    };
+
+    $scope.valid = function() {
+        if ($scope.trigger == null || $scope.method == null) {
```
Comment (john-m-liu) [inline public/static/js/subscriptions.js]: (!$scope.trigger || !$scope.method) to check for undefined

File: public/static/js/subscriptions.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        return false;
+    };
+
+    $scope.method = {};
+    $scope.targets = {
+        SUBSCRIPTION_EVERGREEN_WEBHOOK: {
+            secret: $scope.generateSecret(),
+        },
+    };
+    if ($scope.c.subscription !== undefined) {
```
Comment (john-m-liu) [inline public/static/js/subscriptions.js]: (!$scope.c.subscription) to check for null

File: public/static/thirdparty/angular-material-mocks.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
+/**
```
[the remark this answers is not in the document]
Comment (richardsamuels) [reply public/static/thirdparty/angular-material-mocks.js]: Nope, I copied the dist straight out of the build. Unless you want them left in, I'll remove them

File: public/static/js/projects.js (hunk 2 above)
Comment (richardsamuels) [reply public/static/js/projects.js]: It's needed when rendering one of the partial templates (subscriptions_list.html). This variable makes the build-break notifications check box show up. I'm going to reuse this template when I implement the subscriptions menu for subscribing to versions/builds/etc.

**Comment by richardsamuels** [comment]
@john-m-liu  Fixed that issue.

**Pull request merged**